src/ray/raylet/node_manager.cc

```diff
@@ -300,9 +401,55 @@ void NodeManager::ScheduleTasks() {
 }
 
 void NodeManager::SubmitTask(const Task &task, const Lineage &uncommitted_lineage) {
+  const TaskSpecification &spec = task.GetTaskSpecification();
+
   // Add the task and its uncommitted lineage to the lineage cache.
   lineage_cache_.AddWaitingTask(task, uncommitted_lineage);
-  // Queue the task according to the availability of its arguments.
+
+  if (spec.IsActorTask()) {
+    // Check whether we know the location of the actor.
+    const auto actor_entry = actor_registry_.find(spec.ActorId());
+    if (actor_entry != actor_registry_.end()) {
+      // We have a known location for the actor.
+      auto node_manager_id = actor_entry->second.GetNodeManagerId();
+      if (node_manager_id == gcs_client_->client_table().GetLocalClientId()) {
+        // The actor is local. Queue the task for local execution.
+        QueueTask(task);
+      } else {
+        // The actor is remote. Forward the task to the node manager that owns
+        // the actor.
+        // TODO(swang): Handle forward task failure.
+        RAY_CHECK_OK(ForwardTask(task, node_manager_id));
+      }
+    } else {
+      // We do not have a registered location for the object, so either the
+      // actor has not yet been created or we missed the notification for the
+      // actor creation. Look up the actor's registered location in case we
+      // missed the creation notification.
```

Comment:
When would we miss the creation notification? E.g., if a node joins late or something? It seems like it will be very easy for this code path to not get tested. What do you think?

Reply:
Or will this actually be pretty likely to get called on the first actor method?

Reply:
Yes, I think it should only happen in the case where a node joins after the actor has already been published. It won't get tested in a single-node setting, unfortunately. I think we should keep it, though, since it's necessary for things like autoscaling. I can add a note about testing it in a cluster setting.